src/storage/src/source/statistics.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Helpers for managing storage statistics.
+
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use timely::progress::frontier::Antichain;
+use timely::progress::Timestamp;
+
+use mz_repr::GlobalId;
+use mz_storage_client::client::SourceStatisticsUpdate;
+
+/// A helper struct designed to make it easy for operators to update metrics
+#[derive(Clone)]
+pub struct SourceStatistics {
+    // We just use `SourceStatisticsUpdate` for convenience here!
+    // The boolean its an initialization flag, to ensure we
+    // don't report a false `snapshot_committed` value before
+    // the `persist_sink` reports the current shard upper.
+    stats: Rc<RefCell<(bool, SourceStatisticsUpdate)>>,
```

Comment:
Still not sure I like that lone `bool` here. Maybe this could be a `Option<SourceStatisticsUpdate>`? But I also don't want to block merging this.